c-api/compat-5.3.c

```diff
@@ -336,22 +392,155 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 
 
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
+  const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__))
+    // use strerror_r here, because it's available on these specific platforms
+    char buf[512];
+    strerror_r(en, buf, 1024);
+    s = buf;
+#elif defined(_MSC_VER)
```

Comment:
This should be a check for `_WIN32`, not the compiler in use.

Reply:
@daurnimator Are you sure all compilers and versions on windows support strerror_s? msvc and the latest mingw does, but on mingw you would probably use strerror_r.